core/src/main/scala/org/http4s/Query.scala

```diff
@@ -212,6 +213,19 @@ object Query {
         case Left(_) => Vector.empty
       }
 
+  /* query       = *( pchar / "/" / "?" )
+   *
+   * These are illegal, but common in the wild.  We will be
+   * "conservative in our sending behavior and liberal in our
+   * receiving behavior", and encode them.
+   */
+  private[http4s] def parser: Parser[Query] = {
```

Comment:
This was a `val` in the original PR, there were some initialization ordering issues that caused a NPE

Reply:
This is one of the dangerous of putting the parsers on the companions.  We don't get compiler help with the ordering of eager vals.  I'm not sure how heavy creating a new `Parser` instance every time is.  We possibly want a lazy val here, or to understand the cycle that caused this.  But this gets us moving forward.